cmd/kube-controller-manager/app/core.go

```diff
@@ -81,21 +82,39 @@ func startServiceController(ctx ControllerContext) (http.Handler, bool, error) {
 }
 
 func startNodeIpamController(ctx ControllerContext) (http.Handler, bool, error) {
-	var clusterCIDR *net.IPNet
 	var serviceCIDR *net.IPNet
+	var err error
+	clusterCIDRs := make([]*net.IPNet, 0)
 
 	if !ctx.ComponentConfig.KubeCloudShared.AllocateNodeCIDRs {
 		return nil, false, nil
 	}
 
-	var err error
+	// cluster cidr list processing
+	//TODO (khenidak) clustercidr processing is the same
+	// across ipam and route and should be unified
 	if len(strings.TrimSpace(ctx.ComponentConfig.KubeCloudShared.ClusterCIDR)) != 0 {
-		_, clusterCIDR, err = net.ParseCIDR(ctx.ComponentConfig.KubeCloudShared.ClusterCIDR)
-		if err != nil {
-			klog.Warningf("Unsuccessful parsing of cluster CIDR %v: %v", ctx.ComponentConfig.KubeCloudShared.ClusterCIDR, err)
+		clusterCIDRList := strings.Split(ctx.ComponentConfig.KubeCloudShared.ClusterCIDR, ",")
+		for idx, thisCIDR := range clusterCIDRList {
+			if idx > 0 && !utilfeature.DefaultFeatureGate.Enabled(kubefeatures.IPv6DualStack) {
+				klog.Warningf("multiple cluster cidrs were supplied and dual stack is off, ignoring all but first")
+				break
+			}
+			if idx > 1 {
+				klog.Warningf("dual stacks enabled but more than 2 cidrs were supplied. ignoring all but first two cidrs")
+				break
+			}
+
+			_, clusterCIDR, err := net.ParseCIDR(thisCIDR)
```

Comment:
should all of this happen before the 2 conditionals so the cidrs are appended before breaking out of the loop